core/src/main/scala/kafka/server/FinalizedFeatureChangeListener.scala

```diff
@@ -143,10 +142,13 @@ class FinalizedFeatureChangeListener(zkClient: KafkaZkClient) extends Logging {
   private class ChangeNotificationProcessorThread(name: String) extends ShutdownableThread(name = name) {
     override def doWork(): Unit = {
       try {
-        ignoring(classOf[InterruptedException]) {
-          queue.take.updateLatestOrThrow()
-        }
+        queue.take.updateLatestOrThrow()
       } catch {
+        case _: InterruptedException =>
+          // We ignore InterruptedException. While the queue is empty and this thread is blocking on
```

Comment:
I think we can just propagate this exception to ShutdownableThread. It ignores this exception if the thread is stopped.

Reply:
Done. Great point.